front-end/src/app/services/app-service.service.spec.ts

```diff
@@ -1,20 +1,33 @@
 import { TestBed } from '@angular/core/testing';
-
+import { HttpClientTestingModule }
+        from '@angular/common/http/testing';
 import { AppServiceService } from './app-service.service';
+import { HttpClient } from '@angular/common/http';
 
 describe('AppServiceService', () => {
   let service: AppServiceService;
+  let httpClient: HttpClient;
 
   beforeEach(() => {
-    TestBed.configureTestingModule({});
+    TestBed.configureTestingModule({
+      imports: [ HttpClientTestingModule ]
+    });
+    httpClient = TestBed.inject(HttpClient);
     service = TestBed.inject(AppServiceService);
   });
 
   it('should be created', () => {
     expect(service).toBeTruthy();
   });
+
+  it('test HttpClient getData', () => {
+    const testData = {"test":"API works!"};
+    httpClient.get('http://localhost:3000/api/getData')
+      .subscribe(data =>
+        expect(data).toEqual(testData)
```

Comment:
It's a great progress.
But your test is "naze" :) 
Because you are still not testing your function "getData"
You need to call "service.getData"
Remove all this (httpClient.get etc...)
Here the solution for this exercise I asked you
You need to have something like this
```ts
const testData = {"test":"API works!"};
service.getData().subscribe(value => expect(value).toEqual(testData));
const req = httpTestingController.expectOne(service.baseUrl+'/api/getData');
req.flush(testData);
```

Look again the link I send to you if needed
https://angular.io/guide/http#testing-http-requests